odk1-src/form-audit-log.rst

```diff
@@ -0,0 +1,146 @@
++-------------------+------------------------------------------------------------------+-------+----------------+
+| form exit         | Exit the form                                                    | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| form resume       | Resume the form                                                  | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| form finalize     | Finalize the form                                                | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| save error        | Error trying to save                                             | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| finalize error    | Error trying to finalize the form (probably encryption related)  | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+| constraint error  | Constraint or required error on finalize                         | No    | ``start`` only |
++-------------------+------------------------------------------------------------------+-------+----------------+
+
+.. _audit-timestamps:
+
+Timestamps
+-----------
+
+Device time can be changed by the user unless Android access controls are put in place. To reduce the ways that audits can be manipulated by modifying device time, clock time is collected once at form launch and all following times are computed relative to that initial reading. This means that while the actual time may potentially be inaccurate, the time elapsed within and between screens is always accurate within one form editing session.
+
+Using epoch time makes it easy to compute elapsed time by subtracting start from end. For example, given the following log:
+
+.. csv-table:: audit.csv
+  :header: event, node, start, end
+
+  form start, , 1488761807863, 
+  question, /data/name, 1488761807868, 1488761809157
+
+The enumerator spent ``1488761809157 - 1488761807868 = 1289`` milliseconds on the screen showing the ``/data/name`` question. This corresponds to ``1289 / 1000 = 1.289`` seconds.
+
+To convert from epoch time to time in UTC in most common spreadsheet programs, divide the epoch time by 86400000 ms per day and add 25569 days between January 1, 1900 (what most spreadsheet programs use as "day zero") and January 1, 1970. For example, to convert the timestamp ``1488761807868``:
+
+.. code-block:: xml
+
+  (1488761807868 / 86400000) + 25569 = 42800.03944
+
+When the cell is set to type ``date time`` in common spreadsheet programs, it will show ``3/6/2017 0:56:48 UTC``.
```

Comment:
So, if I was viewing this in Excel, I'd probably add a column of formula cells doing that conversion math, *and then* have that result viewed as `date/time`?

Reply:
That seems likely. I can make that explicit.